src/common/util/submit-data.js

```diff
@@ -78,15 +55,24 @@ submitData.xhr = function xhr ({ url, body = null, sync, method = 'POST', header
 }
 
 /**
- * Send by appending an <img> element to the page. Do NOT call this function outside of a guaranteed web window environment.
- * @param {Object} args - The args
- * @param {string} args.url - The URL to send to
- * @returns {HTMLImageElement}
+ * Send via fetch with keepalive true
+ * @param {Object} args - The args.
+ * @param {string} args.url - The URL to send to.
+ * @param {string=} args.body - The Stringified body.
+ * @param {string=} [args.method=POST] - The XHR method to use.
+ * @param {{key: string, value: string}[]} [args.headers] - The headers to attach.
+ * @returns {XMLHttpRequest}
  */
-submitData.img = function img ({ url }) {
-  var element = new Image()
-  element.src = url
-  return element
+export function fetchKeepAlive ({ url, body = null, method = 'POST', headers = [{ key: 'content-type', value: 'text/plain' }] }) {
+  return fetch(url, {
+    method,
+    headers: headers.reduce((aggregator, header) => {
+      aggregator.push([header.key, header.value])
```

Comment:
Does `headers` work with an array of arrays here?  I've always just seen it as a flat object of k:v defs so just double checking.

Reply:
Yup: https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers